src/discof/restore/fd_snaprd_tile.c

```diff
@@ -227,42 +255,55 @@ read_file_data( fd_snaprd_tile_t *  ctx,
     return;
   }
 
-  fd_stem_publish( stem, 0UL, FD_SNAPSHOT_MSG_DATA, ctx->out.chunk, (ulong)result, 0UL, 0UL, 0UL );
-  ctx->out.chunk = fd_dcache_compact_next( ctx->out.chunk, (ulong)result, ctx->out.chunk0, ctx->out.wmark );
+  fd_stem_publish( stem, ctx->out_snapctl.idx, FD_SNAPSHOT_MSG_DATA, ctx->out_snapctl.chunk, (ulong)result, 0UL, 0UL, 0UL );
+  ctx->out_snapctl.chunk = fd_dcache_compact_next( ctx->out_snapctl.chunk, (ulong)result, ctx->out_snapctl.chunk0, ctx->out_snapctl.wmark );
 }
 
 static void
 read_http_data( fd_snaprd_tile_t *  ctx,
                 fd_stem_context_t * stem,
                 long                now ) {
-  uchar * out = fd_chunk_to_laddr( ctx->out.wksp, ctx->out.chunk );
+  uchar * out = fd_chunk_to_laddr( (void *)ctx->out_snapctl.chunk0, ctx->out_snapctl.chunk );
 
   ulong buffer_avail = fd_ulong_if( -1!=ctx->local_out.dir_fd, SNAPRD_FILE_BUF_SZ-ctx->local_out.write_buffer_len, ULONG_MAX );
-  ulong data_len = fd_ulong_min( buffer_avail, ctx->out.mtu );
+  ulong data_len = fd_ulong_min( buffer_avail, ctx->out_snapctl.mtu );
   int result = fd_sshttp_advance( ctx->sshttp, &data_len, out, now );
 
+  char const * full_snapshot_name;
+  char const * incremental_snapshot_name;
+  fd_sshttp_snapshot_names( ctx->sshttp, &full_snapshot_name, &incremental_snapshot_name );
+  char snapshot_path[ PATH_MAX+30UL ]; /* 30 is fd_cstr_nlen( "https://255.255.255.255:12345/", ULONG_MAX ) */
```

Comment:
```suggestion
  char snapshot_path[ PATH_MAX+30UL ]; /* 30 is fd_cstr_nlen( "https://255.255.255.255:65536/", ULONG_MAX ) */
```
a little clearer intent